lib/elixir/lib/kernel/cli.ex

```diff
@@ -216,6 +218,12 @@ defmodule Kernel.CLI do
 
   # Parse shared options
 
+  defp parse_shared([opt | t], _config) when opt in @standalone_opts and t != [],
+    do: print_standalone_error(opt)
+
+  defp parse_shared([opt | _t], config) when opt in @standalone_opts and config != @blank_config,
+    do: print_standalone_error(opt)
+
   defp parse_shared([opt | _t], _config) when opt in ["-v", "--version"] do
```

Comment:
Move this clause with the version check to the stop:

```elixir
def main([opt]) when opt in ["-v", "--version"] do
  ...
```

This way, you can simplify the print_standalone_error check to be:

```suggestion
  defp parse_shared([opt | _], _config) when opt in @standalone_opts,
    do: print_standalone_error(opt)

  defp parse_shared([opt | _t], _config) when opt in ["-v", "--version"] do
```

Reply:
Actually, nevermind. This won't work for `iex --version` because the args are `--no-halt +iex --version`. Now I am thinking we should NOT make `--version` standalone and change it to no longer automatically exit.